src/typesystem.rs

```diff
@@ -1029,12 +1036,35 @@ pub fn is_same_type_class(
 ) -> bool {
     let ltype = index.find_intrinsic_type(ltype);
     let rtype = index.find_intrinsic_type(rtype);
+
     match ltype {
         DataTypeInformation::Integer { .. } => matches!(rtype, DataTypeInformation::Integer { .. }),
         DataTypeInformation::Float { .. } => matches!(rtype, DataTypeInformation::Float { .. }),
         DataTypeInformation::String { encoding: lenc, .. } => {
             matches!(rtype, DataTypeInformation::String { encoding, .. } if encoding == lenc)
         }
+
+        // We have to handle 3 different cases here:
+        // 1. foo := ADR(bar)
+        // 2. foo := REF(bar)
+        // 3. foo := &bar
+        DataTypeInformation::Pointer { .. } => match rtype {
+            // Case 1: ADR(bar) returns a LWORD value, thus check if we're working with a LWORD
```

Comment:
I know we talked about LWORD before, but I think we could do this better
the Integer{...} has a size inside, and the POINTER_SIZE constant is correct here, or will be updated once we change pointer sizes for targets < 64bit
We can leave the ADR returning an LWORD, but we could already be checking on the correct size

Reply:
Even better would be the index.type_layout().p64.bits() here which is the pointer size for the target but we can keep this for later once we support non-64bit systems.